### DISCUSSION TITLE
Should crc vm forcefully stopped after 90sec as most of cloud platform offer?

### DISCUSSION THREAD

**Discussion by praveenkumar** [crc-org/crc#3209] [General]
Recently with 4.8 testing one of the blocking issue we found is vm taking longer time to shutdown gracefully (~7-8 mins) and a bug is filled against it https://bugzilla.redhat.com/show_bug.cgi?id=1965992 but when we are looking at a cloud provider (gcp) to reproduce same issue, noticed that the stop there just wait for around 90 sec and if VM doesn't shutdown gracefully it just forceful shut it off. 

This is what we do right now if this happen after 2 mins and we as user to forcefully shut it down. Should we also follow same path, instead of 120 sec we choose 90 sec window to do force shutdown?

**Comment by gbraad** [comment]
90secs to wait is quite long... and I am OK to force it, as long as it does
not cause corruption to the disk

**Comment by cfergeau** [comment]
Not fully sure what the issue is suggesting
1) to offer forceful shutdown after 90s instead of the current 120s?
2) to always do forceful shutdown after 90s without asking the user?
3) to do both 1. and 2. ?

**Comment by praveenkumar** [comment]
@cfergeau 3rd option but in case of 4.8 since it is always going to be the case so may be 2, I am open for discussion. Ideally this bug should be fixed and vm should gracefully shutdown around during that time but it is not happening as of now :(

**Comment by cfergeau** [comment]
We have both `-f` and an interactive prompt to force forceful shutdown after 90s, 2. would make `-f` obsolete. How does one indicate to GCP that they do not want a forceful shutdown after 90s? I'm surprised they do this by default :)

**Comment by praveenkumar** [comment]
> How does one indicate to GCP that they do not want a forceful shutdown after 90s? I'm surprised they do this by default :)

@cfergeau As soon as user click on `stop` they will get a pop up `Stop shuts down the instance. If the shutdown doesn't complete within 90 seconds, the instance is forced to halt. This can lead to file-system corruption. Do you want to stop instance`
